Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py

```diff
@@ -172,18 +178,33 @@ def PyInit(self):
 
     def PyExec(self):
         data = self._expand_groups()
+
         bkg = self.getProperty("BackgroundWorkspace").valueAsStr  # same background for all
         cal = self.getProperty("CalibrationWorkspace").value  # same calibration for all
         numberBins = self.getProperty("NumberBins").value
         outWS = self.getPropertyValue("OutputWorkspace")
         summing = self.getProperty("Sum").value  # [Yes or No]
+        filtered_eve = self.getProperty("FilteredInput").value  # [Yes or No]
 
         # convert all of the input workspaces into spectrum of "target" units (generally angle)
         data, masks = self._convert_data(data)
 
         # determine x-range
         xMin, xMax = self._locate_global_xlimit(data)
 
+        if filtered_eve:
+            ExtractMask(InputWorkspace=cal, OutputWorkspace="mask_shared", EnableLogging=False)
+            if cal is not None:
```

Comment:
General coding practice is that rather than having 
```
if not value:
  # do something
else:
  # do something else
```
the branch should be flipped to
```
if value:
  # do something
else:
  # do something else
```

Reply:
thanks a lot @peterfpeterson for your comments! I will adapt those changes and push to the PR soon.